app/javascript/controllers/notifications_controller.js

```diff
@@ -60,7 +60,8 @@ export default class extends Controller {
   }
 
   get #serviceWorkerRegistration() {
-    return navigator.serviceWorker.getRegistration(window.location.host)
+    const hostAndProtocol = window.location.protocol + '//' + window.location.host;
+    return navigator.serviceWorker.getRegistration(hostAndProtocol)
```

Comment:
```suggestion
    return navigator.serviceWorker.getRegistration(window.location.origin)
```

https://developer.mozilla.org/en-US/docs/Web/API/Location/origin